### DISCUSSION TITLE
Help with my 500's (plural)

### DISCUSSION THREAD

**Discussion by dgtl** [GerryDazoo/Slinger#348] [Q&A]
This process is a bit beyond my skill set. I can not even determine how you're accessing the UI of the Slingbox's software.  I did at lease catch the passwords for all my devices before the servers went dark. I'd assume you're getting hit up majorly on this work around, so I can only be hopeful that you may see and respond to my plea for help. Like as in baby-steps.

**Comment by GerryDazoo** [comment]
This is a standalone program it does not access any UI. Download the software, read the readme. edit the config.ini with the password for one of your slingboxes and turn on that slingbox. Run the program, watch your slingbox.
Read the supplied release notes on how to set it up for multiple boxes.

**Comment by tryin100** [comment]
> This process is a bit beyond my skill set. I can not even determine how you're accessing the UI of the Slingbox's software. I did at lease catch the passwords for all my devices before the servers went dark. I'd assume you're getting hit up majorly on this work around, so I can only be hopeful that you may see and respond to my plea for help. Like as in baby-steps.

i just went through this with multiple 500's and also needed help to use this absolutely fantastic 'slinger' program.  let me know if you are still baffled.

**Comment by dgtl** [comment]
I don mean to bug, but I can't quiet figure this out. I missed my slingbox.

Not sure what file  or files to download from your gethub.

**Reply by Toei79** [reply]
Depends  if windows or linux

**Comment by dgtl** [comment]
Windows

**Reply by DoogieHowitzerWMD** [reply]
Understand that whatever computer you choose to set up and run the code will be the source of your video streams. That computer will need to be on whenever you want to watch the content coming from your slingboxes. It can be in the same location as the slingboxes (this is preferable) or it can be at a different location. You can view the video streams on any device that is able to connect to that computer (also the computer itself).

Assuming that the windows PC is still what you want to use to run the server, the first thing is to download the repository of all the code here. The easiest way to do that on windows is just downloading the zip folder.

Next, create a folder somewhere on your C drive (you can put it in your documents folder or something, but the simplest thing is to make it its own directory so `C:\Slingbox` or `C:\Slinger`... whatever you want to name it). Extract everything from the zipped folder and move it to the new folder you created.

Once you get to that point, the needs of your particular setup will determine how to configure things in the next steps. We need to know:
- Whether the windows PC will be at the same location as the slingboxes themselves, or at a different location
- Whether you are looking to watch video streams at locations other than where the windows PC will be located

**Comment by tryin100** [comment]
dgtl

I posted a pdf with step by step instructions for installing multiple sling 500s for the 'non expert'
look for this discussion topic "Step by Step guide for non expert to Install Slinger for Multiple 500"

**Comment by dgtl** [comment]
I am stuck, this is what i get when i run the server:

C:\Slinger>goto :loop

C:\Slinger>(
REM if code is not on your C drive switch to the correct drive letter
 REM g:
 REM use the full pathname to where your code is located
 cd slingbox
 slingbox_server.exe
 REM or if your using python use the full path to whatever python
interpreter you're using.
 REM "c:\Program Files\Python39\python.exe" slingbox_server.py
)
The system cannot find the path specified.
Version : 3.08f Running on Windows-10-10.0.22000-SP0 pid= 31136
slingbox_server.exe
Using config file config.ini
Traceback (most recent call last):
  File "slingbox_server.py", line 1452, in <module>
Exception in thread Thread-1 (ConnectionManager):
Traceback (most recent call last):
  File "threading.py", line 1016, in _bootstrap_inner
  File "threading.py", line 953, in run
    cp = ConfigParser()
  File "configparser.py", line 698, in read
  File "slingbox_server.py", line 1061, in ConnectionManager
  File "configparser.py", line 1097, in _read
    cp = ConfigParser()
  File "configparser.py", line 698, in read
configparser.DuplicateOptionError: While reading from 'config.ini' [line
70]: option 'videosource' in section '500' already exists
[31136] Failed to execute script 'slingbox_server' due to unhandled
exception!
  File "configparser.py", line 1097, in _read
configparser.DuplicateOptionError: While reading from 'config.ini' [line
70]: option 'videosource' in section '500' already exists

and here is my config.ini file

[SLINGBOX]
sbtype=500
;sbtype="ProHD"
;sbtype="240"
password=epBh8sIAZoyA4MB

;uncomment the next lines ipaddress and port with your local network info
;if you don't want the server to automatically find your
;slingbox on the local network. Needed if server and slingbox are
;not on the same LAN segment. Also removes requirement for the netifaces
module
;ipaddress=192.168.86.81
;port=5201

[500]
sbtype=500
password=epBh8sIAZoyA4MB
finderid=4E2C0AA3BF9A146637EAE0940D7FB0E8
ipaddress=192.168.86.81
port=5201
Resolution=12
FrameRate=60
VideoBandwidth=2000
VideoSmoothness=63
VideoSource=1
maxstreams=3
StartChannel=155
;RemoteLock=yes
Remote=500_remote.txt

;Replace with your finderid, If and only if you need remote access and
don't
;have a static ip and you plan on using my service to access you server
remotely.
;Please read the release notes. Most people don't need this.
;finderid=4E2C0AA3BF9A146637EAE0940D7FB0E8

;If you've got more than one slingbox set name. THe server will use this
when generating
;logs to make it easier to see what's going on.
;name=something

;Following are the default values
;Default resolution when server starts. You can change this and reconnect
without having to restart the server
;valid range = 1..15
;Resolution=12

;Valid Values 1, 6, 10, 15, 20, 30, 60   depending on resolution
;FrameRate=30

;Min 50 Max 8000
;VideoBandwidth=2000

;Min 0 (auto) max 63
;VideoSmoothness=63

;Send Iframe every n seconds Max 30
;IframeRate=5

;Audio Bit Rate. Valid Options 16, 20, 32, 40, 48, 64, 96 Default 32
;AudioBitRate=32

;Video Source  0, 1, 2, 3 depending on your hardware corresponds to one of
;Composite, Component, S-Video, HDMI or Tuner.
;Pro:        0=(Analog)Tuner   1=Composite   2=S-Video     3=Component (Via
HDMI to Component Cable)
;ProHD:      0=Tuner           1=Composite   2=S-Video     3=Component
;Solo:       0=Composite       1=S-Video     2=Component
;500:        0=Composite       1=Component   2=HDMI
;350/M1/M2:  0=Composite       1=Component
; If you don't set this, the code will use the last configured input
; I recommend setting this value. If not set correctly, you'll often only
see a black screen.
; Note: Setting invalid options may cause unexpected hardware behavior. For
example, setting
;  2 or 3 for M2 hardware causes it to reboot.
VideoSource = 1

[SERVER]
; local port number for the server to listen on for connections
; Send all streaming and Remote request to this port number.
port=65432
maxremotestreams=2
URLbase=Gerry
enableremote=yes

[REMOTE]
;see release notes for changing this if your remote doesn't work, But as a
start make it
;the same as your configured VideoSource 0-3. See above.
code=1
;path to an external remote control definitions file, if the defaults
;aren't working for you or you want to change the format of the web page.
;use the supplied remote.txt as a starting point.
;include=remote.txt


On Fri, Dec 2, 2022 at 6:18 AM DoogieHowitzerWMD ***@***.***>
wrote:

> Understand that whatever computer you choose to set up and run the code
> will be the source of your video streams. That computer will need to be on
> whenever you want to watch the content coming from your slingboxes. It can
> be in the same location as the slingboxes (this is preferable) or it can be
> at a different location. You can view the video streams on any device that
> is able to connect to that computer (also the computer itself).
>
> Assuming that the windows PC is still what you want to use to run the
> server, the first thing is to download the repository of all the code here.
> The easiest way to do that on windows is just downloading the zip folder.
>
> Next, create a folder somewhere on your C drive (you can put it in your
> documents folder or something, but the simplest thing is to make it its own
> directory so C:\Slingbox or C:\Slinger... whatever you want to name it).
> Extract everything from the zipped folder and move it to the new folder you
> created.
>
> Once you get to that point, the needs of your particular setup will
> determine how to configure things in the next steps. We need to know:
>
>    - Whether the windows PC will be at the same location as the
>    slingboxes themselves, or at a different location
>    - Whether you are looking to watch video streams at locations other
>    than where the windows PC will be located
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/GerryDazoo/Slinger/discussions/348#discussioncomment-4292875>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGDYB63QSA3PJ7P7FS3SJNLWLIAK5ANCNFSM6AAAAAASI2OSHI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
<+13233096994>

**Reply by GerryDazoo** [reply]
You need to work on your cutting and pasting skills. You've completely confused the "unified config.ini file format and the old config.ini
Change [SLINGBOX] to [SLINGBOXES]
Delete the [REMOTE] section
Delete everything in the [SLINGBOXES] section. Add back in sb1=500

**Reply by GerryDazoo** [reply]
Or Move everything  in the [500] section to [SLINGBOX]

**Reply by DoogieHowitzerWMD** [reply]
> File "configparser.py", line 698, in read
configparser.DuplicateOptionError: While reading from 'config.ini' [line
70]: option 'videosource' in section '500' already exists
[31136] Failed to execute script 'slingbox_server' due to unhandled
exception!

You have video source specified twice in your config file. Your config file is pretty messy overall though. It looks like you combine stuff from the unified_config example with the generic config file.

Your config file should look more like this: 

`[SLINGBOXES]
sb1=500
;sb2="ProHD"
;sb3="240"

[500]
sbtype="350/500/M1"
password=epBh8sIAZoyA4MB
finderid=4E2C0AA3BF9A146637EAE0940D7FB0E8
ipaddress=192.168.86.81
port=5201
Resolution=12
FrameRate=60
VideoBandwidth=2000
VideoSmoothness=63
VideoSource=1
maxstreams=3
StartChannel=155
;RemoteLock=yes
Remote=500_remote.txt

[SERVER]
; local port number for the server to listen on for connections
; Send all streaming and Remote request to this port number.
port=65432
maxremotestreams=3
URLbase=Gerry
enableremote=yes`

**EDIT:** Fixed error in config file code

**Reply by GerryDazoo** [reply]
[SLINGBOXES] not [SLINGBOX]

**Reply by DoogieHowitzerWMD** [reply]
Whoops... Good catch Gerry.

That's what I get for not having copied everything straight from the unified_config file for that edit.

**Comment by dgtl** [comment]
I promise, I promise this is my last time bugging you. If i don't figure it
out from here, I will just have to concede to living life without my
slingbox.

this is what runs in my command screen when i double click the "RunSling -
Windows Batch File":

C:\Slinger>goto :loop

C:\Slinger>(
REM if code is not on your C drive switch to the correct drive letter
 REM g:
 REM use the full pathname to where your code is located
 cd slingbox
 slingbox_server.exe
 REM or if your using python use the full path to whatever python
interpreter you're using.
 REM "c:\Program Files\Python39\python.exe" slingbox_server.py
)
The system cannot find the path specified.
Version : 3.08f Running on Windows-10-10.0.22000-SP0 pid= 32156
slingbox_server.exe
Using config file config.ini
Exception in thread Thread-1 (ConnectionManager):
Traceback (most recent call last):
Traceback (most recent call last):
  File "slingbox_server.py", line 1452, in <module>
  File "threading.py", line 1016, in _bootstrap_inner
    cp = ConfigParser()
  File "configparser.py", line 698, in read
  File "threading.py", line 953, in run
  File "configparser.py", line 1086, in _read
  File "slingbox_server.py", line 1061, in ConnectionManager
configparser.MissingSectionHeaderError: File contains no section headers.
file: 'config.ini', line: 1
'`[SLINGBOXES]\n'
[32156] Failed to execute script 'slingbox_server' due to unhandled
exception!
    cp = ConfigParser()
  File "configparser.py", line 698, in read
  File "configparser.py", line 1086, in _read
configparser.MissingSectionHeaderError: File contains no section headers.
file: 'config.ini', line: 1
'`[SLINGBOXES]\n'

Do I need to move anything into the highlighted directory?

this is my "config.ini" file located inside directory location  "C:slinger"

`[SLINGBOXES]
sbtype=500

[500]
sbtype=500
password=epBh8sIAZoyA4MB
finderid=4E2C0AA3BF9A146637EAE0940D7FB0E8
ipaddress=192.168.86.81
port=5201
Resolution=12
FrameRate=60
VideoBandwidth=2000
VideoSmoothness=63
VideoSource=1
maxstreams=3
StartChannel=155
;RemoteLock=yes
Remote=500_remote.txt

[SERVER]
; local port number for the server to listen on for connections
; Send all streaming and Remote request to this port number.
port=65432
maxremotestreams=3
URLbase=Gerry
enableremote=yes`


On Mon, Dec 5, 2022 at 8:12 AM GerryDazoo ***@***.***> wrote:

> [SLINGBOXES] not [SLINGBOX]
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/GerryDazoo/Slinger/discussions/348#discussioncomment-4314842>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGDYB6446UOTIVGIQYJLZ2DWLYH6HANCNFSM6AAAAAASI2OSHI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
<+13233096994>

**Reply by DoogieHowitzerWMD** [reply]
> File "configparser.py", line 1086, in _read
configparser.MissingSectionHeaderError: File contains no section headers.
file: 'config.ini', line: 1
'`[SLINGBOXES]\n'

I think you have an extra ' at the start of the text in your config file.

It looks like my previous post was not formatted properly, so if you copied that one, tha'ts where the ' came from.

The verbatim config file should look like this (I've double checked the formatting this time):

```
[SLINGBOXES]
sbtype=500

[500]
sbtype=500
password=epBh8sIAZoyA4MB
finderid=4E2C0AA3BF9A146637EAE0940D7FB0E8
ipaddress=192.168.86.81
port=5201
Resolution=12
FrameRate=60
VideoBandwidth=2000
VideoSmoothness=63
VideoSource=1
maxstreams=3
StartChannel=155
;RemoteLock=yes
Remote=500_remote.txt

[SERVER]
; local port number for the server to listen on for connections
; Send all streaming and Remote request to this port number.
port=65432
maxremotestreams=3
URLbase=Gerry
enableremote=yes
```

It looks like there was an extra ' at the end of the very last line as well. Remove the one at the beginning and the one at the end (or copy-paste the one above) and you should be good. If not, we'll take another look.

**Comment by dgtl** [comment]
Okay, seen to have gotten some success. I got rid of all  '  in the .ini
file, this is who i get now in command.


C:\Slinger>(
REM if code is not on your C drive switch to the correct drive letter
 REM g:
 REM use the full pathname to where your code is located
 cd slingbox
 slingbox_server.exe
 REM or if your using python use the full path to whatever python
interpreter you're using.
 REM "c:\Program Files\Python39\python.exe" slingbox_server.py
)
The system cannot find the path specified.
Version : 3.08f Running on Windows-10-10.0.22000-SP0 pid= 7632
slingbox_server.exe
Using config file config.ini
Connection Manager Running on port 65432 with 3 max streams using URL Gerry.
BOXES [('sbtype', '500')]
starting up on  port 65432 8388608
Streamer Running:  3 config.ini 500 500 65432 8388608
Checking for slingbox at ('192.168.86.81', 5201)
Building page for 500
Reading Custom Remote definition from 500_remote.txt
Error reading remote definition file, using defaults
 * Serving Flask app 'slingbox_server' (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production
deployment.
   Use a production WSGI server instead.
 * Debug mode: off
12/06/2022, 10:18:34.199  Registering Slingboxes
{'4E2C0AA3BF9A146637EAE0940D7FB0E8': 65432}
Getting external IP address
Redirector, my external IP address 47.149.132.142
Error connecting to  ('192.168.86.81', 5201)
500 No valid slingbox ip info found in config.ini
Finding Slingbox on local network. My IP Info =  192.168.86.81
Slingbox Found 192.168.86.248 5401 " Casa Lawndale " FinderID
E22195123E979146745F5511908C32B3
Slingbox Found 192.168.86.249 5204 " SlingTiVO " FinderID
9FE571525E15604A070F4CDEF6337A0D
Slingbox Found 192.168.86.59 5208 " Bedroom, Master " FinderID
89A15ABD59279317D7C8C6BF63691DB4
12/06/2022, 10:18:36.397  Slinger Remote Access Test failure
('47.149.132.142', 65432) [WinError 10061] No connection could be made
because the target machine actively refused it
No working external portmaps. Not registering slingboxes
Retry 1 in 10 minutes
Found more than one slingbox on the local network.
Please select the one you want to use and update the config.ini accordingly.

500 Giving up. Sorry..


On Tue, Dec 6, 2022 at 9:45 AM DoogieHowitzerWMD ***@***.***>
wrote:

> File "configparser.py", line 1086, in _read
> configparser.MissingSectionHeaderError: File contains no section headers.
> file: 'config.ini', line: 1
> '`[SLINGBOXES]\n'
>
> I think you have an extra ' at the start of the text in your config file.
>
> If you copied the one I pasted, it might be due to how github use text
> enclosed in ' to format things to look like code. I probably added an extra
> in there.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/GerryDazoo/Slinger/discussions/348#discussioncomment-4325535>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGDYB623N4EXNRV3XHR5KR3WL53UHANCNFSM6AAAAAASI2OSHI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
<+13233096994>

**Reply by DoogieHowitzerWMD** [reply]
Looks like the server found three slingboxes on your network, but the config file doesn't have an IP address that matches any of those three.

Do you know which of the three slingboxes you have the admin password for in the config file? If you are you trying to get all three slingboxes on the server, you will need to modify the config file to account for two additional slingboxes.

**Comment by dgtl** [comment]
I just want one slingbox on the server for now. Here is my password file,
it got a bit messy.

var
sling_account_boxes={"memberslingbox":{"4e2c0aa3bf9a146637eae0940d7fb0e8":{"lookupByFinderId":true,"adminPassword":"epBh8sIAZoyA4MB","userPassword":"nSCoh1B0zWZ01MH","finderId":"4E2C0AA3BF9A146637EAE0940D7FB0E8","isOwner":true,"productSignature":18,"passwordAutoMode":true,"displayName":"Living
Room","memberSlingBoxId":3528481},"9fe571525e15604a070f4cdef6337a0d":
{"lookupByFinderId":true,"adminPassword":"99gIqoxsZGIVA3C","userPassword":"MqHdTiQWi8j2Jrm","finderId":"9FE571525E15604A070F4CDEF6337A0D","isOwner":true,"productSignature":18,"passwordAutoMode":true,"displayName":"SlingTiVO","memberSlingBoxId":3524792},"89a15abd59279317d7c8c6bf63691db4":{"lookupByFinderId":true,"adminPassword":"h5g5ToEPSgtJjTC","userPassword":"lAzFPBENVrBtkzQ","finderId":"89A15ABD59279317D7C8C6BF63691DB4","isOwner":true,"productSignature":18,"passwordAutoMode":true,"displayName":"Bedroom,
Master","memberSlingBoxId":3302314}},"size":3}


The highlighted is the one i'm trying to get on the server.

On Tue, Dec 6, 2022 at 10:57 AM DoogieHowitzerWMD ***@***.***>
wrote:

> Looks like the server found three slingboxes on your network, but the
> config file doesn't have an IP address that matches any of those three.
>
> Do you know which of the three slingboxes you have the admin password for
> in the config file? If you are you trying to get all three slingboxes on
> the server, you will need to modify the config file to account for two
> additional slingboxes.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/GerryDazoo/Slinger/discussions/348#discussioncomment-4326139>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGDYB6ZLXPXPJ5BWD6WVBJDWL6EAXANCNFSM6AAAAAASI2OSHI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
<+13233096994>

**Reply by DoogieHowitzerWMD** [reply]
So a few things...

From these three slingboxes that the server found:

```
Slingbox Found 192.168.86.248 5401 " Casa Lawndale " FinderID
E22195123E979146745F5511908C32B3
Slingbox Found 192.168.86.249 5204 " SlingTiVO " FinderID
9FE571525E15604A070F4CDEF6337A0D
Slingbox Found 192.168.86.59 5208 " Bedroom, Master " FinderID
89A15ABD59279317D7C8C6BF63691DB4
```

I don't see one in the file with the info for you slingboxes with this finderID: `4E2C0AA3BF9A146637EAE0940D7FB0E8`

The other two match in both files, so I'd suspect that those two would both be more likely to work. which is why I put all three together in the one config file:

```
[SLINGBOXES]
sbtype=Livingroom
sbtype=SlingTiVo
sbtype=Bedroom

[Livingroom]
sbtype="350/500/M1"
password=epBh8sIAZoyA4MB
finderid=4E2C0AA3BF9A146637EAE0940D7FB0E8
ipaddress=192.168.86.248
port=5401
Resolution=12
FrameRate=60
VideoBandwidth=2000
VideoSmoothness=63
VideoSource=1
maxstreams=3
StartChannel=155
;RemoteLock=yes
Remote=remote.txt

[SlingTiVo]
sbtype="350/500/M1"
password=99gIqoxsZGIVA3C
finderid=9FE571525E15604A070F4CDEF6337A0D
ipaddress=192.168.86.249
port=5204
Resolution=12
FrameRate=60
VideoBandwidth=2000
VideoSmoothness=63
VideoSource=1
maxstreams=3
StartChannel=155
;RemoteLock=yes
Remote=remote.txt

[Bedroom]
sbtype="350/500/M1"
password=h5g5ToEPSgtJjTC
finderid=89A15ABD59279317D7C8C6BF63691DB4
ipaddress=192.168.86.59
port=5208
Resolution=12
FrameRate=60
VideoBandwidth=2000
VideoSmoothness=63
VideoSource=1
maxstreams=3
StartChannel=155
;RemoteLock=yes
Remote=remote.txt

[SERVER]
; local port number for the server to listen on for connections
; Send all streaming and Remote request to this port number.
port=65432
maxremotestreams=3
URLbase=Gerry
enableremote=yes
```

I also see these lines here at the end:

```
('47.149.132.142', 65432) [WinError 10061] No connection could be made
because the target machine actively refused it
No working external portmaps. Not registering slingboxes
```

The IP address you had in the config file `192.168.86.81` doesn't correspond to what the server output lists for the IP addresses of your slingboxes (was .81 for the computer your using to run the server?), but it might also be that port 65432 is not open on your router for the machine running your server. If you're looking to watch the slingboxes away from your home, that is something you'll need to do.

**Comment by dgtl** [comment]
tivo one

On Tue, Dec 6, 2022 at 11:46 AM DoogieHowitzerWMD ***@***.***>
wrote:

> I don't see anything highlighted.
>
> For now, are you looking to start with the bedroom one, the tivo one, or
> the living room one?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/GerryDazoo/Slinger/discussions/348#discussioncomment-4326528>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGDYB6YBNFZWIUCTWUNTQ3LWL6JXVANCNFSM6AAAAAASI2OSHI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
<+13233096994>

**Reply by DoogieHowitzerWMD** [reply]
The new config file from my last comment https://github.com/GerryDazoo/Slinger/discussions/348#discussioncomment-4326739 should work with the TiVo one. Give it a try.

**Comment by dgtl** [comment]
okay, looks like it got the server part running, now trying to
configure the VLC


C:\Slinger>(
REM if code is not on your C drive switch to the correct drive letter
 REM g:
 REM use the full pathname to where your code is located
 cd slingbox
 slingbox_server.exe
 REM or if your using python use the full path to whatever python
interpreter you're using.
 REM "c:\Program Files\Python39\python.exe" slingbox_server.py
)
The system cannot find the path specified.
Version : 3.08f Running on Windows-10-10.0.22000-SP0 pid= 23804
slingbox_server.exe
Using config file config.ini
Connection Manager Running on port 65432 with 3 max streams using URL Gerry.
BOXES [('sbtype', 'SlingTiVo')]
Streamer Running:  3 config.ini SlingTiVo SlingTiVo 65432 8388608
starting up on  port 65432 8388608
Checking for slingbox at ('192.168.86.249', 5204)
('192.168.86.249', 5204) OK
SlingTiVo Using slingbox at  ('192.168.86.249', 5204)
Streamer:  SlingTiVo Waiting for first stream, flushing any IR requests
that arrive while not connected to slingbox
Building page for SlingTiVo
Reading Custom Remote definition from remote.txt
 * Serving Flask app 'slingbox_server' (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production
deployment.
   Use a production WSGI server instead.
 * Debug mode: off
12/06/2022, 12:54:13.497  Registering Slingboxes
{'9FE571525E15604A070F4CDEF6337A0D': 65432}
Getting external IP address
Redirector, my external IP address 47.149.132.142
Registering 9FE571525E15604A070F4CDEF6337A0D:65432

Register Successful


On Tue, Dec 6, 2022 at 12:41 PM DoogieHowitzerWMD ***@***.***>
wrote:

> The new config file from my last comment should work with the TiVo one.
> Give it a try.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/GerryDazoo/Slinger/discussions/348#discussioncomment-4326846>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGDYB65SKXZ5XYLXFHBIFHDWL6QHXANCNFSM6AAAAAASI2OSHI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
<+13233096994>

**Reply by DoogieHowitzerWMD** [reply]
Yeah, that looks good for the TiVo slingbox.

Internally, the stream address should be `http://127.0.0.1:65432/Gerry/SlingTiVo` or `http://192.168.86.###:65432/Gerry/SlingTiVo` with ### being the remaining portion of the IP address for the computer running the server. I would try VLC on the computer itself, and on your phone.

To access it externally (outside of your home network), you want `http://47.149.132.142:65432/Gerry/SlingTivo`. Eventually, you would need to set up a DDNS to create a web address that is always synched up with what your home IP is, as it's possible for it to change at some point.

**Comment by dgtl** [comment]
Thank you, thank you, thank you, shoot me your Patreon or paypal so I can
buy you dinner.

oh, ...no mobile access remote access?

On Tue, Dec 6, 2022 at 1:39 PM DoogieHowitzerWMD ***@***.***>
wrote:

> Yeah, that looks good for the TiVo slingbox.
>
> Internally, the stream address should be
> http://127.0.0.1:65432/Gerry/SlingTiVo or
> http://192.168.86.###:65432/Gerry/SlingTiVo with ### being the remaining
> portion of the IP address for the computer running the server. I would try
> VLC on the computer itself, and on your phone.
>
> To access it externally (outside of your home network), you want
> http://47.149.132.142:65432/Gerry/SlingTivo. Eventually, you would need
> to set up a DDNS to create a web address that is always synched up with
> what your home IP is, as it's possible for it to change at some point.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/GerryDazoo/Slinger/discussions/348#discussioncomment-4327198>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGDYB62QE4WYGV37CFZ6VETWL6XAFANCNFSM6AAAAAASI2OSHI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
<+13233096994>

**Reply by DoogieHowitzerWMD** [reply]
Glad it's working for you.

Is it only working on your network and not outside of it?

**Comment by dgtl** [comment]
That's correct inside only

**Reply by DoogieHowitzerWMD** [reply]
That likely means you need to open external port 65432 on your router for the computer running the server.

**Comment by dgtl** [comment]
Okay

**Comment by dgtl** [comment]
I got it to play externally, but how do I control the channel, I have a
TiVo that sources the signal.

And I was serious about buying your dinner. Send me your patreon, paypal or
other means to receive $$$, and I'll send over enough money for you to buy
dinner.

On Tue, Dec 6, 2022 at 3:13 PM Mike Dear ***@***.***> wrote:

> Okay
>
> On Tue, Dec 6, 2022, 15:04 DoogieHowitzerWMD ***@***.***>
> wrote:
>
>> That likely means you need to open external port 65432 on your router for
>> the computer running the server.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/GerryDazoo/Slinger/discussions/348#discussioncomment-4327659>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AGDYB6YHYU6A22WPOUXPZJTWL7BBNANCNFSM6AAAAAASI2OSHI>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***>
>>
>

-- 
<+13233096994>

**Reply by GerryDazoo** [reply]
There is a Tivo remote definitions file  in the CustomRemotes folder. But the default remote should let you change channels.
Point any ol' browser to the 
http://127.0.0.1:65432/Remote/SlingTiVo

Replace 127.0.0.1 with 47.149.132.142 if you want to connect remotely

**Reply by DoogieHowitzerWMD** [reply]
To change the channel, you access the remote for your particular slingbox through any web browser:

`http://47.149.132.142:65432/Remote/SlingTiVo`

You put the name of the particular slingbox (listed in your config file) at the end of the address, after Remote/ 

You will probably have the default remote file. You can find fancier remote files that people have put in the custom remotes folder, but I use the default one, because I pretty much only change the channel numbers and do little else with it.

One caveat is that you need at least one stream active in order for the remote to work (so, if you are trying to watch on your phone and change the channel, it's not gonna work if you have to close the stream to open the webpage). To get around that, you'll have to pass the channel code through the initial address for the stream:

`http://47.149.132.142:65432/Gerry/SlingTivo?channel=###`

where ### is the channel number (it can be any number of digits)

And don't worry about compensating me. I've gotten help here from Gerry and from other folks on many occasions. I'm just paying it forward.

**Comment by dgtl** [comment]
Hello,
Now i'm being greedy, ...you guys think there is a way I can get the
slinger server running on a microsoft 2022 server. I already deployed it,
and the devices actually register, just can not get it to connect and play
from the video client. If you want to take a stab at it, I'll let the
survey run.

This is the configuration I made for it.

[SLINGBOXES]
sb1=SlingTiVo
sb2=LivingRoom

[SlingTiVo]
sbtype="350/500/M1"
password=99gIqoxsZGIVA3C
finderid=9FE571525E15604A070F4CDEF6337A0D
ipaddress=192.168.86.249
port=5201
Resolution=12
FrameRate=60
VideoBandwidth=2000
VideoSmoothness=63
VideoSource=1
maxstreams=3
;StartChannel=618
;RemoteLock=yes
Remote=500_remote.txt

[LivingRoom]
sbtype="350/500/M1"
password=epBh8sIAZoyA4MB
finderid=4E2C0AA3BF9A146637EAE0940D7FB0E8
ipaddress=192.168.86.47
port=5204
Resolution=12
FrameRate=60
VideoBandwidth=2000
VideoSmoothness=63
VideoSource=1
maxstreams=3
;StartChannel=618
;RemoteLock=yes
Remote=500_remote.txt


[SERVER]
; local port number for the server to listen on for connections
; Send all streaming and Remote requests to this port number.
port=65432
maxremotestreams=3
URLbase=Gerry
enableremote=yes

and my server's IP address is: 172.81.60.53  I don't know how to locate the
server's actual port number.

On Wed, Dec 7, 2022 at 11:54 AM DoogieHowitzerWMD ***@***.***>
wrote:

> To change the channel, you access the remote for your particular slingbox:
>
> http://47.149.132.142:65432/Remote/SlingTiVo
>
> You put the name of the particular slingbox (listed in your config file)
> at the end of the address, after Remote/
>
> You will probably have the default remote file. You can find fancier
> remote files that people have put in the custom remotes folder, but I use
> the default one, because I pretty much only change the channel numbers and
> do little else with it.
>
> And worry about compensating me. I've gotten help here from Gerry and from
> other folks on many occasions. I'm just paying it forward.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/GerryDazoo/Slinger/discussions/348#discussioncomment-4336695>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGDYB6YAYUJEIZWZ737X3ELWMDTQDANCNFSM6AAAAAASI2OSHI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
<+13233096994>

**Reply by GerryDazoo** [reply]
I'm sure you checked the obvious things.
1) your portmap on your router is pointing to the 2022Server IP and the port number is 65432
2) THe firewall on the 2022Server is not blocking port 65432.

**Reply by flipper2222222** [reply]
I have slinger running on MS 2022 server, I didn't do anything special, however I did change video client from VLC to SM player for better overall results.

**Comment by dgtl** [comment]
Yeah, so the ip addresses in my configuration is my public IP addressand
the port to each sling device  and the two devices are registering.

So they're waiting and ready for me to connect. But how do i figure out the
port on the 2022 msft server?

I thought it would be:

http://72.81.60.53:####/Gerry/ SlingTiVo

The 65432 is not working.

**Reply by GerryDazoo** [reply]
There's no figuring out what the port is. It's port=65432
So did you even check what I suggested?

**Comment by dgtl** [comment]
It not a port mapping issue, because the 2022server registered my sling
device.

**Reply by GerryDazoo** [reply]
I fail to see what that has to do with the port mapping on your router to allow external access to the slingbox_server on your Win Server box. Did you actually go check it or are you assuming it's OK?
Post a screen shot of your router config and the output of the ipconfig command on the server.

**Comment by dgtl** [comment]
Here's a better one.
